bayem/vba.py

```diff
@@ -0,0 +1,519 @@
+import logging
+from dataclasses import dataclass
+from time import perf_counter
+from tabulate import tabulate
+from typing import Union, Dict, Tuple
+
+import numpy as np
+import scipy.special as special
+
+from .distributions import MVN, Gamma
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class Options:
+    tolerance: float = 0.1
+    maxiter: int = 50
+    maxtrials: int = 10
+    update_noise: Union[Dict, bool] = True
+    index_ARD: Tuple[int] = ()
+
+
+class CDF_Jacobian:
+    def __init__(self, f, transformation):
+        self._f = f
+        self._T = transformation
+
+    def __call__(self, _x):
+        x = np.copy(_x)
+
+        for iParam in range(len(x)):
+            eps = 1.0e-7  # approx sqrt(machine precision)
```

Comment:
Would it be possible to have that as a parameter (so e.g. in the constructor with standard e-7? In particular, it might be nice to be able to provide absolute values

Reply:
I added a `cdf_eps` to the `Options` class. 

Also note that this central differences Jacobian idea is IMO a reasonable default value for the _lazy_ user ;)
There are many ways to improve the Jacobian and the user can conveniently provide his/her own implementation. Would you agree?

Reply:
Maybe this could be stressed here, the basic user might not realize how to change that himself.